File: project-euler/0002/0002.js
function solution(a, b, sum) {

    var next = a + b;

    if(next > 4000000) {
        return sum;
    }

    return solution(b, next, (next % 2 === 0) ? sum + next : sum)
}

var result = solution(0, 1, 0);

console.log(result);

File: project-euler/0002/0002.go
package main

func solution(a, b, sum int) int {

	next := a + b
	if next > 4000000 {
		return sum
	}
	if next%2 == 0 {
		return solution(b, next, sum+next)
	}
	return solution(b, next, sum)
}
